File: MiniDOMBase/DMD/Sistema/Users/CUserXGroupCursor.vb
Imports System.Net.Mail 'importo il Namespace
Imports System.Net.Sockets
Imports minidom
Imports minidom.Databases
Imports minidom.Anagrafica

Partial Public Class Sistema

    Public Class CUserXGroupCursor
        Inherits DBObjectCursorBase(Of CUserXGroup)

        Private m_UserID As CCursorField(Of Integer)
        Private m_GroupID As CCursorField(Of Integer)

        Public Sub New()
            Me.m_UserID = New CCursorField(Of Integer)("User")
            Me.m_GroupID = New CCursorField(Of Integer)("Group")
        End Sub

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_UsersXGroup"
        End Function

        Public ReadOnly Property UserID As CCursorField(Of Integer)
            Get
                Return Me.m_UserID
            End Get
        End Property

        Public ReadOnly Property GroupID As CCursorField(Of Integer)
            Get
                Return Me.m_GroupID
            End Get
        End Property

    End Class

End Class

File: MiniDOMBase/DMD/Sistema/Users/UsersXGroup/CUserXGroupCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;


namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Cursore sugli oggetti di tipo <see cref="CUserXGroup"/>
        /// </summary>
        [Serializable]
        public class CUserXGroupCursor
            : minidom.Databases.DBObjectCursorBase<CUserXGroup>
        {
            private DBCursorField<int> m_UserID;
            private DBCursorField<int> m_GroupID;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CUserXGroupCursor()
            {
                m_UserID = new DBCursorField<int>("User");
                m_GroupID = new DBCursorField<int>("Group");
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Sistema.Users.UsersXGroupsRepository;
            }

            /// <summary>
            /// Discriminator
            /// </summary>
            /// <returns></returns>
            public override string GetTableName()
            {
                return "tbl_UsersXGroup";
            }

            /// <summary>
            /// UserID
            /// </summary>
            public DBCursorField<int> UserID
            {
                get
                {
                    return m_UserID;
                }
            }

            /// <summary>
            /// GroupID
            /// </summary>
            public DBCursorField<int> GroupID
            {
                get
                {
                    return m_GroupID;
                }
            }
        }
    }
}